src/ty/fields.rs

```diff
@@ -86,6 +86,17 @@ pub struct Field<T: Form = MetaForm> {
     ty: T::Type,
     /// The name of the type of the field as it appears in the source code.
     type_name: T::String,
+    /// This field should be encode/decoded as a
+    /// [`Compact`](parity_scale_codec::Compact) field
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "is_false", default))]
+    compact: bool,
```

Comment:
I think an even more compact representation for the exact same use case would be to use yet another enum variant in order to represent compacted types. The idea is to add another variant `Compact(TypeDefCompact<T>)` to the `TypeDef` enum: https://docs.rs/scale-info/0.4.1/scale_info/enum.TypeDef.html
The `TypeDefCompact` type internally could look like this:
```rust
pub struct TypeDefCompact<T>
where
    T: Form = MetaForm
{
    #[serde(rename = "type")]
	type_param: T::Type,
}
```
Then instead of requiring each type to have this `bool` field we'd simply have a compact variant only for those type definitions that are actually compactly encoded. Since this is kind of a special case this design would suite much better. Also it would better reflect the encoding structure.

Reply:
At first glance I think this solution is more elegant than having the field, while still making it a first class citizen (unlike just using a type e.g. `Compact<T>`.